aiida_quantumespresso/parsers/pp.py

```diff
@@ -0,0 +1,332 @@
+        Parse 2D GNUPlot formatted output
+
+        :param data_file_str: the data file read in as a single string
+        """
+        data_lines = data_file_str.splitlines()
+
+        coords = []
+        data = []
+
+        for line in data_lines:
+            if line == '':
+                continue
+            else:
+                split_line = line.split()
+                coords.append([float(split_line[0]), float(split_line[1])])
+                data.append(float(split_line[2]))
+
+        coords_array = np.array(coords)
+        data_array = np.array(data)
+
+        arraydata = orm.ArrayData()
+        arraydata.set_array('xy_coordinates', coords_array)
+        arraydata.set_array('data', data_array)
+
+        return arraydata
+
+
+    def parse_gaussian(self, data_file_str):
+        """
+        Parse Gaussian Cube formatted output
+
+        :param data_file_str: the data file read in as a single string
+        """
+
+        lines = data_file_str.splitlines()
+
+        atoms_line = lines[2].split()
+        atoms = int(atoms_line[0]) # The number of atoms listed in the file
+        header = lines[:6+atoms]  # The header of the file: comments, the voxel, and the number of atoms and datapoints
+        data_lines = lines[6+atoms:] # The actual data: atoms and volumetric data
+
+        # Parse the declared dimensions of the volumetric data
+        x_line = header[3].split()
+        xdim = int(x_line[0])
+        y_line = header[4].split()
+        ydim = int(y_line[0])
+        z_line = header[5].split()
+        zdim = int(z_line[0])
+        dimensions_array = np.array([xdim, ydim, zdim])
+
+        #Get the vectors describing the basis voxel
+        voxel_array = np.array(
+            [[x_line[1], x_line[2], x_line[3]],
+            [y_line[1], y_line[2], y_line[3]],
+            [z_line[1], z_line[2], z_line[3]]],
+            dtype=np.float64
+        )
+
+        # Get the volumetric data
+        data_array = np.zeros((xdim, ydim, zdim))
+
+        # The data is organised into columns with a fixed number of values
+        # Gather up all the data points into one list for easier unpacking
+        datalist = []
+        for line in data_lines:
+            for i in range(0, len(line), 13):
+                data_point = line[i:i+13].strip()
+                if data_point != '':
+                    datalist.append(float(data_point))
+
+        # Unpack the list and repack as a 3D array
+        # Note the unusual indexing: cube files run over the z index first, then y and x.
+        # E.g. The first volumetric data point is x,y,z = (0,0,0) and the second is (0,0,1)
+        for i in range(0, xdim):
+            for j in range(0, ydim):
+                for k in range(0, zdim):
+                    data_array[i, j, k] = (datalist[(i*ydim*zdim)+(j*zdim)+k])
+
+        arraydata = orm.ArrayData()
+        arraydata.set_array('voxel', voxel_array)
+        arraydata.set_array('dimensions', dimensions_array)
```

Comment:
I'm not sure if it's necessary to set the `dimensions` explicitly as an array - is there a case where it's different from the shape of the `data` array? Otherwise, it will be the same as the content of the `array|data` attribute.

Reply:
Correct, it can be removed